python/lsst/pipe/tasks/detectAndMeasure.py

```diff
@@ -0,0 +1,255 @@
+        """Construct a DetectAndMeasureTask
+
+        Arguments in addition to the standard Task arguments:
+        @param[in] dataPrefix  prefix for name of source tables;
+            - for calexp use the default of ""
+            - for coadds use coaddName + "Coadd"
+        @param[in,out] schema  schema for sources; if None then one is constructed
+        """
+        pipeBase.Task.__init__(self, **kwargs)
+        self.dataPrefix = dataPrefix
+        self.algMetadata = dafBase.PropertyList()
+        if schema is None:
+            schema = SourceTable.makeMinimalSchema()
+        self.schema = schema
+        self.makeSubtask("detection", schema=self.schema)
+        if self.config.doDeblend:
+            self.makeSubtask("deblend", schema=self.schema)
+        self.makeSubtask("measurement", schema=self.schema, algMetadata=self.algMetadata)
+        if self.config.doMeasureApCorr:
+            # add field to flag stars useful for measuring aperture correction
+            self.makeSubtask("measureApCorr", schema=schema)
+
+    @pipeBase.timeMethod
+    def run(self, exposure, exposureIdInfo, background=None, allowApCorr=True):
+        """!Detect, deblend and perform single-frame measurement on sources and refine the background model
+
+        @param[in,out] exposure  exposure to process. Background must already be subtracted
+            to reasonable accuracy, else detection will fail.
+            The background model is refined and resubtracted.
+            apCorrMap is set if measuring aperture correction.
+        @param[in]     exposureIdInfo  ID info for exposure (an lsst.daf.butlerUtils.ExposureIdInfo)
+        @param[in,out] background  background model to be modified (an lsst.afw.math.BackgroundList),
+            or None to create a new background model
+        @param[in] allowApCorr  allow measuring and applying aperture correction?
+
+        @return pipe_base Struct containing these fields:
+        - exposure: input exposure (as modified in the course of runing)
+        - sourceCat: source catalog (an lsst.afw.table.SourceTable)
+        - background: background model (input as modified, or a new model if input is None);
+            an lsst.afw.math.BackgroundList
+        """
+        if background is None:
+            background = BackgroundList()
+
+        sourceIdFactory = IdFactory.makeSource(exposureIdInfo.expId, exposureIdInfo.unusedBits)
+
+        table = SourceTable.make(self.schema, sourceIdFactory)
+        table.setMetadata(self.algMetadata)
+
+        detRes = self.detection.run(table=table, exposure=exposure, doSmooth=True)
+        sourceCat = detRes.sources
+        if detRes.fpSets.background:
+            background.append(detRes.fpSets.background)
+
+        if self.config.doDeblend:
+            self.deblend.run(
+                exposure = exposure,
+                sources = sourceCat,
+                psf = exposure.getPsf(),
+            )
```

Comment:
This style makes the code look much longer than it needs to.  Can't you just put the kwargs all on the same line?